### DISCUSSION TITLE
Running cargo-make without cargo-metadata

### DISCUSSION THREAD

**Discussion by agrif** [sagiegurari/cargo-make#1076] [Q&A]
Hello! I'm trying to use cargo-make to generate all of the source code for my crate.

However, cargo-make sees that there is a Cargo.toml file in my package root, and tries to run `cargo metadata`, which fails (since there are no source files yet.) Is there any way to configure cargo-make to ignore the Cargo.toml file? I notice that cargo-make runs just fine *without* a Cargo.toml, and only fails if one exists that it can't read metadata from.

For the time being, I'm working around this by including a single empty example binary, but I was wondering if there was a better way.

**Comment by sagiegurari** [comment]
i use cargo-metadata to understand the crate (especially workspaces). without it some key info is missing on how to run it.
but i see your point, and will think of something.

**Comment by sagiegurari** [comment]
@agrif just released a version of cargo-make that should support this flow

**Reply by agrif** [reply]
Thank you!
